### PR TITLE
Issue 34 - Quit PhhamtomJS browser after taking screenshot

### LINKED ISSUE #34: Multiple Phantomjs process left running
After take screenshot and crop the image, the phantomjs process is left running in the server.

The soluction would be follow the tips in this stack overflow http://stackoverflow.com/questions/25110624/how-to-properly-stop-phantomjs-execution

### LINKED ISSUE #36: Webdriver exception at find_element_by_id method
At one customer this exception is raised:

```python
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.CMFFormController.FSControllerPythonScript, line 105, in __call__
  Module Products.CMFFormController.Script, line 145, in __call__
  Module Products.CMFCore.FSPythonScript, line 127, in __call__
  Module Shared.DC.Scripts.Bindings, line 322, in __call__
  Module Shared.DC.Scripts.Bindings, line 359, in _bindAndExec
  Module Products.PythonScripts.PythonScript, line 344, in _exec
  Module script, line 47, in content_status_modify
   - <FSControllerPythonScript at /Plone/pitacos/se-gritar-pega-ladrao-1/content_status_modify>
   - Line 47
  Module Products.CMFCore.WorkflowTool, line 241, in doActionFor
  Module Products.CMFCore.WorkflowTool, line 570, in _invokeWithNotification
  Module zope.event, line 31, in notify
  Module zope.component.event, line 24, in dispatch
  Module zope.component._api, line 136, in subscribers
  Module zope.component.registry, line 321, in subscribers
  Module zope.interface.adapter, line 585, in subscribers
  Module zope.component.event, line 32, in objectEventNotify
  Module zope.component._api, line 136, in subscribers
  Module zope.component.registry, line 321, in subscribers
  Module zope.interface.adapter, line 585, in subscribers
  Module collective.behavior.featuredimage.subscriber, line 45, in update_featuredimage
  Module collective.behavior.featuredimage.subscriber, line 18, in _get_screenshot
  Module selenium.webdriver.remote.webdriver, line 269, in find_element_by_id
  Module selenium.webdriver.remote.webdriver, line 745, in find_element
  Module selenium.webdriver.remote.webdriver, line 236, in execute
  Module selenium.webdriver.remote.errorhandler, line 194, in check_response
NoSuchElementException: Message: {"errorMessage":"Unable to find element with id 'featuredimage'","request":{"headers":{"Accept":"application/json","Accept-Encoding":"identity","Connection":"close","Content-Length":"94","Content-Type":"application/json;charset=UTF-8","Host":"127.0.0.1:52797","User-Agent":"Python-urllib/2.7"},"httpVersion":"1.1","method":"POST","post":"{\"using\": \"id\", \"sessionId\": \"d59a70d0-bbd7-11e6-a016-37225ccd7a44\", \"value\": \"featuredimage\"}","url":"/element","urlParsed":{"anchor":"","query":"","file":"element","directory":"/","path":"/element","relative":"/element","port":"","host":"","password":"","user":"","userInfo":"","authority":"","protocol":"","source":"/element","queryKey":{},"chunks":["element"]},"urlOriginal":"/session/d59a70d0-bbd7-11e6-a016-37225ccd7a44/element"}}
Screenshot: available via screen
```

The strange part is that the image is generated without problems.

It looks like we need to wait for the page or phantomjs be ready to select the element, here are some sources about the same issue:

* http://stackoverflow.com/questions/27599633/getting-error-in-click-method-of-selenium-in-python-when-i-find-the-element-fro
* http://stackoverflow.com/questions/30344223/exception-on-execution-phantomjs-selenium-webdriver-python-program
* http://selenium-python.readthedocs.io/waits.html

### PR DESCRIPTION
**Pull request by rodfersou**
closes #34

### CODE REVIEW

File: src/collective/behavior/featuredimage/subscriber.py [hunk 1]
```diff
@@ -15,9 +19,17 @@ def _get_screenshot(page):
...
-    size = el.size
+    try:
+        el = WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.ID, 'featuredimage'))
+        )
+        # Put these lines after driver.quit will raise URLError exception
+        location = el.location
+        size = el.size
+    finally:
+        driver.close()
```
Comment (hvelarde) [inline src/collective/behavior/featuredimage/subscriber.py]: you don't need `driver.close()`, that's automatically executed by `driver.quit()`; also, I'm not pretty sure of what are you doing in the next line, and why are you doing that.

Comment (rodfersou) [reply src/collective/behavior/featuredimage/subscriber.py]: please read this http://stackoverflow.com/a/38493285

Comment (hvelarde) [reply src/collective/behavior/featuredimage/subscriber.py]: so, the `driver.close()` is not needed as I said.

**Comment by rodfersou** [comment]
@hvelarde any doubts on this PR? can we merge it?

File: src/collective/behavior/featuredimage/subscriber.py [hunk 2]
```diff
@@ -15,9 +19,16 @@ def _get_screenshot(page):
...
     # crop image
-    el = driver.find_element_by_id('featuredimage')
-    location = el.location
-    size = el.size
+    try:
+        el = WebDriverWait(driver, 10).until(
+            EC.presence_of_element_located((By.ID, 'featuredimage'))
+        )
+        # Put these lines after driver.quit will raise URLError exception
+        location = el.location
```
Comment (hvelarde) [inline src/collective/behavior/featuredimage/subscriber.py]: if the previous line fails then we will have an exception below this block; we need to review this.

Comment (rodfersou) [reply src/collective/behavior/featuredimage/subscriber.py]: agree

File: src/collective/behavior/featuredimage/subscriber.py [hunk 3]
```diff
@@ -15,9 +19,16 @@ def _get_screenshot(page):
...
+        size = el.size
+    finally:
+        driver.service.process.send_signal(signal.SIGTERM)
+        driver.quit()
```
Comment (rodfersou) [inline src/collective/behavior/featuredimage/subscriber.py]: how about to add a simple **return** here?

Comment (hvelarde) [reply src/collective/behavior/featuredimage/subscriber.py]: finally runs always

**Pull request merged**